From 32f3c4536f40fc89ccd8cca14a0b85cb2f954e58 Mon Sep 17 00:00:00 2001
From: Pavankumar Nandeshwar <quic_pnandesh@quicinc.com>
Date: Sat, 24 Aug 2024 09:09:58 +0530
Subject: [PATCH] QSDK: ath12k: Allow ds setup for peer on a non-ds ab in an
 mlo

Even though an ab does not support direct switch, but when it is
part of an mlo, and when the associated station is connected
in mlo mode, allow ds peer setup on the corresponding peer
to make sure that vp number is set.

Patch-dependency: 900-a-wifi-ath12k-clean-up-bonded-model.patch
Signed-off-by: Pavankumar Nandeshwar <quic_pnandesh@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/ppe.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/ppe.c
+++ b/drivers/net/wireless/ath/ath12k/ppe.c
@@ -542,18 +542,13 @@ void ath12k_dp_peer_ppeds_route_setup(st
 	u32 priority_valid = 0, src_info = ahsta->ppe_vp_num;
 	struct ieee80211_sta *sta;
 
-	if (!test_bit(ATH12K_FLAG_PPE_DS_ENABLED, &ab->dev_flags))
-		return;
-
 	if (arvif->ahvif->vdev_type == WMI_VDEV_TYPE_MONITOR ||
 	    (arvif->ahvif->vdev_type == WMI_VDEV_TYPE_AP &&
 	     arvif->vdev_subtype == WMI_VDEV_SUBTYPE_MESH_11S))
 	     	return;
 
-	if (ahvif->ppe_vp_num == -1 ||
-	    ahvif->ppe_vp_type != PPE_VP_USER_TYPE_DS) {
-		ath12k_dbg(ab, ATH12K_DBG_PPE, "DS is disabled for peer %pM\n",
-			   arsta->addr);
+	if (ahvif->ppe_vp_num == -1) {
+		ath12k_dbg(ab, ATH12K_DBG_PPE, "Invalid ppe vp number\n");
 		return;
 	}
 
diff --git a/drivers/net/wireless/ath/ath12k/dp.c b/drivers/net/wireless/ath/ath12k/dp.c
index 4f577c3..23b8c2f 100644
--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -1456,9 +1456,6 @@ static int ath12k_dp_ppe_rxole_rxdma_cfg(struct ath12k_base *ab)
 	struct ath12k_dp_htt_rxdma_ppe_cfg_param param = {0};
 	int ret;
 
-	if (!test_bit(ATH12K_FLAG_PPE_DS_ENABLED, &ab->dev_flags))
-		return 0;
-
 	param.override = 1;
 	param.reo_dst_ind = HAL_REO2PPE_DST_IND;
 	param.multi_buffer_msdu_override_en = 0;
